build/opencover-report.ps1

```diff
@@ -24,32 +29,121 @@ $packageConfig = [xml](Get-Content ..\.nuget\packages.config)
+$pdb2pdb_version = $packageConfig.SelectSingleNode('/packages/package[@id="Microsoft.DiaSymReader.Pdb2Pdb"]').version
 
 $packages_folder = '..\packages'
 $opencover_console = "$packages_folder\OpenCover.$opencover_version\tools\OpenCover.Console.exe"
 $xunit_runner_console = "$packages_folder\xunit.runner.console.$xunitrunner_version\tools\xunit.console.x86.exe"
 $report_generator = "$packages_folder\ReportGenerator.$reportgenerator_version\tools\ReportGenerator.exe"
+$pdb2pdb = "$packages_folder\Microsoft.DiaSymReader.Pdb2Pdb.$pdb2pdb_version\tools\Pdb2Pdb.exe"
 $report_folder = '.\OpenCover.Reports'
-$target_dll = "..\StyleCop.Analyzers\StyleCop.Analyzers.Test\bin\$Configuration\StyleCop.Analyzers.Test.dll"
+$symbols_folder = '.\OpenCover.Symbols'
+$target_dll = "..\StyleCop.Analyzers\StyleCop.Analyzers.Test\bin\$Configuration\net452\StyleCop.Analyzers.Test.dll"
+$target_dll_csharp7 = "..\StyleCop.Analyzers\StyleCop.Analyzers.Test.CSharp7\bin\$Configuration\net46\StyleCop.Analyzers.Test.CSharp7.dll"
+
+If (Test-Path $symbols_folder) {
+	Remove-Item -Recurse -Force $symbols_folder
+}
+
+$symbols_folder_csharp6 = Join-Path $symbols_folder 'CSharp6'
+$symbols_folder_csharp7 = Join-Path $symbols_folder 'CSharp7'
+mkdir $symbols_folder | Out-Null
+mkdir $symbols_folder_csharp6 | Out-Null
+mkdir $symbols_folder_csharp7 | Out-Null
+
+function Convert-Coverage-Pdb() {
+	param (
+		[string]$assembly,
+		[string]$outputDir
+	)
+	$sourceDir = [IO.Path]::GetDirectoryName($assembly)
+	$outputName = [IO.Path]::ChangeExtension([IO.Path]::GetFileName($assembly), 'pdb')
+	$sourcePdb = Join-Path $sourceDir $outputName
+	$output = Join-Path $outputDir $outputName
+	if (Test-Path $sourcePdb) {
+		&$pdb2pdb $assembly /out $output
+
+		# Workaround for https://github.com/OpenCover/opencover/issues/800
+		Remove-Item $sourcePdb
+		Copy-Item $assembly $outputDir
+	}
+}
+
+function Extract-Coverage-Pdb() {
+	param (
+		[string]$assembly,
+		[string]$outputDir
+	)
+	$sourceDir = [IO.Path]::GetDirectoryName($assembly)
+	$outputName = [IO.Path]::ChangeExtension([IO.Path]::GetFileName($assembly), 'pdb')
+	$intermediatePdb = Join-Path $sourceDir $outputName
+	$output = Join-Path $outputDir $outputName
+	if (-not (Test-Path $output)) {
+		&$pdb2pdb $assembly /out $intermediatePdb /extract
+		if (Test-Path $intermediatePdb) {
+			&$pdb2pdb $assembly /pdb $intermediatePdb /out $output
+			Remove-Item $intermediatePdb
+
+			# Workaround for https://github.com/OpenCover/opencover/issues/800
+			Copy-Item $assembly $outputDir
+		}
+	}
+}
+
+$target_dir = [IO.Path]::GetDirectoryName($target_dll)
+Get-ChildItem $target_dir -Filter *.dll | Foreach-Object { Convert-Coverage-Pdb -assembly $_.FullName -outputDir $symbols_folder_csharp6 }
+Get-ChildItem $target_dir -Filter *.dll | Foreach-Object { Extract-Coverage-Pdb -assembly $_.FullName -outputDir $symbols_folder_csharp6 }
```

Comment:
❓ Why is the Extract-Coverage-Pdb step necessary? As far as I can see, due to the changes made all builds will produce PDB files anyway, so the Convert-Coverage-Pdb should be enough right?

Reply:
I'm planning to use this to document a general approach for other repositories already using portable and/or embedded PDBs to adopt OpenCover as part of their build pipelines.